### ISSUE TITLE
Naming rules should allow keywords to be wrapped between backticks

### ISSUE THREAD

**Issue by dimasadryantos** [pinterest/ktlint#2352]
## Expected Behavior
Function name with backticks and short name should not throwing an error when formatting.

## Observed Behavior
Ktlint detect function name with backticks wrongly for example if I have method like in the following : 

```
fun `when`(): String {
    return ""
}
```
when I format the code it would throw an error like the following : 
 `[ktlint] /Users/dsetyawan/Documents/pAPI/spring-kotlin-demo/src/main/kotlin/com/example/springkotlindemo/SpringKotlinDemoApplication.kt:13:5: Function name should start with a lowercase letter (except factory methods) and use camel case (standard:function-naming)`

## Steps to Reproduce
Clone my example project : https://github.com/dimasadryantos/spring-kotlin-demo/tree/master
Run : `mvn clean compile` or `ktlint --format `

## Your Environment

* Version of ktlint used: 0.51.0-FINAL
* Operating System and version: Mac
* Detailed in dependencies provided in pom.xml https://github.com/dimasadryantos/spring-kotlin-demo/blob/master/pom.xml

**Comment by paul-dingemans** [comment]
Duplicate of #2339

**Comment by JakeWharton** [comment]
This is not a duplicate of that issue. You cannot remove the backticks here as the name is otherwise a keyword. Backticks are required to use spaces but it is certainly not their only use.

**Comment by paul-dingemans** [comment]
> This is not a duplicate of that issue. You cannot remove the backticks here as the name is otherwise a keyword. Backticks are required to use spaces but it is certainly not their only use.

Indeed, you can not remove the backticks here. 

According to [Kotlin coding conventions](https://kotlinlang.org/docs/coding-conventions.html#names-for-test-methods):
> In tests (and only in tests), you can use method names with spaces enclosed in backticks.

So when you read above literally, it would allow the example code above as a kind of helper method inside a Test class. But, from the Kotlin coding conventions it is quite clear that it is meant for functions annotated with `@Test`.

**Comment by JakeWharton** [comment]
Right. So this issue should be reopened as there are valid, non-test-based uses of backticks. If you are overriding a Java method whose name is a Kotlin keyword, or you need to provide binary compatibility with a previous version of a library which used a Kotlin keyword as a function or property name they are necessary and should not be flagged.

**Comment by dimasadryantos** [comment]
I'd agree with @JakeWharton , as this issue is only appear on this version.

**Comment by JakeWharton** [comment]
In my specific case, I have a property named "package":
```kotlin
val `package` = type.names[0]
```
It represents a package, so that's what I named it.

The failing rule says

> Property name should start with a lowercase letter and use camel case

It _does_ start with a lowercase "p" and use camel case. The backticks are not part of the name. The backticks allow me to tell the compiler's parser that I am producing a variable name and not a hard keyword.

**Comment by paul-dingemans** [comment]
From the view point of the Kotlin parser, the backticks are an integral part of the identifier as can be seen in screenshot below. On the right hand side you the the PSI (AST) structure of the parsed file. The highlighted `IDENTIFIER` field matches with the name of the variable including the backticks.

But I do see the point that you're making that the keywords should be accepted, even in case they are included between backticks. This applies to rules `property-naming`, `function-naming` and `class-naming`.
